lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -297,7 +265,35 @@ void LocalEnforcer::aggregate_records(
           record.rule_id(), record.bytes_tx(), record.bytes_rx(), uc);
     }
   }
-  notify_finish_report_for_sessions(session_map, session_update);
+  complete_termination_for_released_sessions(
+      session_map, sessions_with_active_flows, session_update);
+}
+
+void LocalEnforcer::complete_termination_for_released_sessions(
+    SessionMap& session_map, std::unordered_set<std::string> sessions_with_active_flows,
+    SessionUpdate& session_update) {
+  // Iterate through sessions and notify that report has finished. Terminate any
+  // sessions that can be terminated.
+  std::vector<std::pair<std::string, std::string>> imsi_to_terminate;
+  for (const auto& session_pair : session_map) {
+    const std::string imsi = session_pair.first;
+    for (const auto& session : session_pair.second) {
+      const std::string session_id = session->get_session_id();
+      // If we did not receive a rule record for the session, this means
+      // PipelineD has reported all usage for the session
+      if (session->get_state() == SESSION_RELEASED &&
+          sessions_with_active_flows.find(imsi) == sessions_with_active_flows.end()) {
+        imsi_to_terminate.push_back(std::make_pair(imsi, session_id));
```

Comment:
can't you run the for at line 290 inside here? 
Otherwise, you can create a function for what you have at line 290 to 296